tests/unit/array/mixins/test_find.py

```diff
@@ -632,6 +632,63 @@ def test_redis_category_filter(filter, checker, columns, start_storage):
     assert all([checker(r) for r in results])
 
 
+from math import radians, cos, sin, asin, sqrt
+
+
+def haversine(lon1, lat1, lon2, lat2):
+    """
+    Calculate the great circle distance between two points
+    on the earth (specified in decimal degrees)
+    """
+    lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, lon2, lat2])
+    dlon = lon2 - lon1
+    dlat = lat2 - lat1
+    a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
+    c = 2 * asin(sqrt(a))
+    r = 6371  # average radius of earth, unit km
+    return c * r  # unit km
```

Comment:
use [sklearn.metrics.pairwise](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics.pairwise).haversine_distances instead, we already install sklearn for tests